src/openzeppelin/tests/test_dual721.cairo

```diff
@@ -0,0 +1,502 @@
+    contract_address_const::<50>()
+}
+fn DATA(success: bool) -> Span<felt252> {
+    let mut data = ArrayTrait::new();
+    if success {
+        data.append(SUCCESS);
+    } else {
+        data.append(FAILURE);
+    }
+    data.span()
+}
+
+///
+/// setup
+///
+
+fn setup_snake() -> (DualERC721, IERC721Dispatcher) {
+    let mut calldata = ArrayTrait::new();
+    calldata.append(NAME);
+    calldata.append(SYMBOL);
+    calldata.append(TOKEN_ID().low.into());
+    calldata.append(TOKEN_ID().high.into());
+    set_caller_address(OWNER());
+    let target = utils::deploy(SnakeERC721Mock::TEST_CLASS_HASH, calldata);
+    (DualERC721 { target: target }, IERC721Dispatcher { contract_address: target })
+}
+
+fn setup_camel() -> (DualERC721, IERC721CamelDispatcher) {
+    let mut calldata = ArrayTrait::new();
+    calldata.append(NAME);
+    calldata.append(SYMBOL);
+    calldata.append(TOKEN_ID().low.into());
+    calldata.append(TOKEN_ID().high.into());
+    set_caller_address(OWNER());
+    let target = utils::deploy(CamelERC721Mock::TEST_CLASS_HASH, calldata);
+    (DualERC721 { target: target }, IERC721CamelDispatcher { contract_address: target })
+}
+
+fn setup_non_erc721() -> DualERC721 {
+    let calldata = ArrayTrait::new();
+    let target = utils::deploy(NonERC721::TEST_CLASS_HASH, calldata);
+    DualERC721 { target: target }
+}
+
+fn setup_erc721_panic() -> (DualERC721, DualERC721) {
+    let snake_target = utils::deploy(SnakeERC721Panic::TEST_CLASS_HASH, ArrayTrait::new());
+    let camel_target = utils::deploy(CamelERC721Panic::TEST_CLASS_HASH, ArrayTrait::new());
+    (DualERC721 { target: snake_target }, DualERC721 { target: camel_target })
+}
+
+fn setup_receiver() -> ContractAddress {
+    utils::deploy(ERC721Receiver::TEST_CLASS_HASH, ArrayTrait::new())
+}
+
+///
+/// case agnostic methods
+///
+
+#[test]
+#[available_gas(2000000)]
+fn test_dual_name() {
+    let (snake_dispatcher, _) = setup_snake();
+    let (camel_dispatcher, _) = setup_camel();
+    assert(snake_dispatcher.name() == NAME, 'Should return name');
+    assert(camel_dispatcher.name() == NAME, 'Should return name');
+}
+
+#[test]
+#[available_gas(2000000)]
+#[should_panic(expected: ('ENTRYPOINT_NOT_FOUND', ))]
+fn test_dual_no_name() {
+    let dispatcher = setup_non_erc721();
+    dispatcher.name();
+}
+
+#[test]
+#[available_gas(2000000)]
+#[should_panic(expected: ('Some error', 'ENTRYPOINT_FAILED', ))]
+fn test_dual_name_exists_and_panics() {
+    let (dispatcher, _) = setup_erc721_panic();
+    dispatcher.name();
+}
+
+#[test]
+#[available_gas(2000000)]
+fn test_dual_symbol() {
+    let (snake_dispatcher, _) = setup_snake();
+    let (camel_dispatcher, _) = setup_camel();
+    assert(snake_dispatcher.symbol() == SYMBOL, 'Should return symbol');
+    assert(camel_dispatcher.symbol() == SYMBOL, 'Should return symbol');
+}
+
+#[test]
+#[available_gas(2000000)]
+#[should_panic(expected: ('ENTRYPOINT_NOT_FOUND', ))]
+fn test_dual_no_symbol() {
```

Comment:
What is the objective of these tests? (The one in line 119 as well).

Reply:
https://github.com/OpenZeppelin/cairo-contracts/pull/623/files/2e0a657e8de8b96252b04eb2ec1a9da0b997d8db#r1228977660